### DISCUSSION TITLE
How to set GitLab Self-Hosted as Context Provider?

### DISCUSSION THREAD

**Discussion by Tomakko** [TabbyML/tabby#3366] [Q&A]
Hi all,

I am trying to achieve what is stated in the title. From out private gitlab I generated a personal access token with all priviliges.

In the tabby browser UI, I am unsure what format Instance URL should have, I tried multiple things, e.g.

https://gitlab.companyname.domain
https://myusername@gitlab.companyname.domain
https://myusername@gitlab.companyname.domain/path_to_repo_i_am_interested_in

However, none of these options seem to work. After adding the context provider the status is 'Ready', however no indexing jobs are started and none of my repos can be references in the chat. 

Adding a public Gitlab repo works fine. The Gitlab server is reachable from the Server tabby is running on.

Could anyone guide me in the right direction? Thanks in advance!

**Comment by zwpaper** [comment]
Hi @Tomakko, Thanks for trying Tabby,

according to your description, the first URL you privided should be correct:
```
https://gitlab.companyname.domain/
```

for the reason why no repos can be found, can you make sure you added you repo in GitLab context source like:

![CleanShot X 2024-11-05 14 47 44](https://github.com/user-attachments/assets/c4f95933-f932-4c51-8969-d3495de10b4c)

**Comment by Tomakko** [comment]
Hey @zwpaper ,

thank you so much for your answer. Tbh, I did not see the plus sign, previously.  Now I was able to add my repository from the list!

Thanks again! Looking forward to exploring Tabby :)
